### PR TITLE
Upgrade with-redux example to app router

### PR DESCRIPTION
**Pull request by dvakatsiienko**
### What?

Update **with-redux** example.

### Why?

**with-redux** example have:
- outdated packages
- outdated approaches and relies on **pages** directory

Since **app router** is stable now and is recommended to use, I've updated this example.

### How?

An update includes:
- move example to **app router**
- update **package.json** deps to the latest versions
- modernize jest by switching from **ts-node** to **@swc/jest**
- fix and overhaul **tests**
- modernize **redux** infra
- overhaul example **source code quality**

### CODE REVIEW

**Comment by dvakatsiienko** [comment]
Hi, @leerob @steven-tey 

It seems that Github Actions CI got broken at [Socket Security: Pull Request Alerts](https://github.com/vercel/next.js/pull/49994/checks?check_run_id=13581495697) and it blocks CI pipeline to advance.

This happens for multiple PR's related to **examples** directory. [I've created an issue for that](https://github.com/vercel/next.js/issues/49997).

**Review by leerob — CHANGES_REQUESTED** [review]
It would be great to simplify this example. Remote saas, remove testing, and focus specifically on Next.js + Redux only in the most minimal, yet complete, example.

**Comment by dvakatsiienko** [comment]
Sounds like a good idea, I'll simplify it.

**Comment by dvakatsiienko** [comment]
@leerob I've removed tests and sass, see if all good now.

File: examples/with-redux/package.json [hunk 1]
```diff
@@ -4,29 +4,24 @@
...
   },
   "dependencies": {
-    "@reduxjs/toolkit": "^1.3.6",
-    "next": "latest",
-    "react": "^18.1.0",
-    "react-dom": "^18.1.0",
-    "react-redux": "^7.2.0"
+    "@reduxjs/toolkit": "1.9.5",
+    "classnames": "2.3.2",
+    "next": "13.4.5",
```
Comment (balazsorban44) [inline examples/with-redux/package.json:13]:
```suggestion
    "next": "latest",
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Let's drop `src` as well, to match other examples.

File: examples/with-redux/package.json [hunk 2]
```diff
@@ -4,29 +4,22 @@
...
+    "@reduxjs/toolkit": "1.9.5",
+    "classnames": "2.3.2",
     "next": "latest",
-    "react": "^18.1.0",
-    "react-dom": "^18.1.0",
-    "react-redux": "^7.2.0"
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "react-redux": "8.1.0",
+    "waait": "1.0.5"
```
Comment (balazsorban44) [inline examples/with-redux/package.json:15]: let's drop `waait` in favor of `await new Promise(r => setTimeout(r, 500));
`

Comment (balazsorban44) [inline examples/with-redux/package.json:15]:
```suggestion
    "react-redux": "8.1.0"
```

File: examples/with-redux/src/app/api/identity-count/route.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+/* Core */
+import { NextResponse } from 'next/server'
+import waait from 'waait'
+
+export async function POST(req: Request, res: Response) {
+  const body = await req.json()
+  const { amount = 1 } = body
+
+  // simulate IO latency
+  await waait(500)
```
Comment (balazsorban44) [inline examples/with-redux/src/app/api/identity-count/route.ts:10]:
```suggestion

export async function POST(req: Request, res: Response) {
  const body = await req.json()
  const { amount = 1 } = body

  // simulate IO latency
  await new Promise(r => setTimeout(r, 500))
```

**Review by balazsorban44 — DISMISSED** [review]
Let's not add barrel files (`index.ts` files that are just re-exports), feels like adding complexity instead of reducing it.

File: examples/with-redux/package.json [hunk 4]
```diff
@@ -4,29 +4,21 @@
     "dev": "next",
     "build": "next build",
     "start": "next start",
-    "type-check": "tsc",
-    "test": "jest"
   },
   "dependencies": {
-    "@reduxjs/toolkit": "^1.3.6",
+    "@reduxjs/toolkit": "1.9.5",
+    "classnames": "2.3.2",
```
Comment (balazsorban44) [inline examples/with-redux/package.json:10]:
```suggestion
```

File: examples/with-redux/src/app/components/Nav.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
+'use client'
+
+/* Core */
+import Link from 'next/link'
+import { usePathname } from 'next/navigation'
+import cx from 'classnames'
```
Comment (balazsorban44) [inline examples/with-redux/src/app/components/Nav.tsx:6]:
```suggestion
```

File: examples/with-redux/src/app/components/Nav.tsx [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+/* Instruments */
+import styles from '../styles/layout.module.css'
+
+export const Nav = () => {
+  const pathname = usePathname()
+
+  return (
+    <nav className={styles.nav}>
+      <Link
+        className={cx(styles.link, { [styles.active]: pathname === '/' })}
```
Comment (balazsorban44) [inline examples/with-redux/src/app/components/Nav.tsx:17]:
```suggestion
        className={`${styles.link} ${pathname === '/' ? styles.active : ''}`}
```

File: examples/with-redux/src/app/components/Nav.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        href="/"
+      >
+        Home
+      </Link>
+      <Link
+        className={cx(styles.link, {
+          [styles.active]: pathname === '/verify',
+        })}
```
Comment (balazsorban44) [inline examples/with-redux/src/app/components/Nav.tsx:25]:
```suggestion
        className={`${styles.link} ${pathname === '/verify' ? styles.active : ''}}
```

**Comment by dvakatsiienko** [comment]
@balazsorban44 removed `src` folder, and barrel files.

**Review by leerob — APPROVED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: eec78254223b0c8817f3e588b224afdee56442a3

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: eec78254223b0c8817f3e588b224afdee56442a3

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Pull request merged**